internal/global/meter.go

```diff
@@ -515,9 +511,33 @@ type registration struct {
 	unregMu sync.Mutex
 }
 
-func (c *registration) setDelegate(m metric.Meter) {
-	insts := unwrapInstruments(c.instruments)
+type unwrapObs struct {
+	embedded.Observer
+	obs metric.Observer
+}
+
+func (uo *unwrapObs) ObserveFloat64(inst metric.Float64Observable, value float64, opts ...metric.ObserveOption) {
+	if un, ok := inst.(unwrapper); ok {
+		inst = un.unwrap().(metric.Float64Observable)
```

Comment:
I'm glad to improve this. I believe the condition we're talking about is impossible without a logic bug internal to this package, since the thing that was wrapped originally by an instrument of the matching type. I'm worried about constructing error conditions that are impossible to test.

I figure that we probably don't need to create a new error condition. If the instrument fails to unwrap as expected, the receiving SDK's asynchronous code path is obligated to handle an error, as there's not much it can do with a reference to the internal/global-private wrapper object.

How does this look, to you?

```
// unwrapFloat64Observable returns an expected metric.Float64Observable after
// unwrapping the global object.
func unwrapFloat64Observable(inst metric.Float64Observable) metric.Float64Observable {
	if unwrapped, ok := inst.(unwrapper); ok {
		if floatObs, ok := unwrapped.unwrap().(metric.Float64Observable); ok {
			// Note: if the unwrapped object does not
			// unwrap as an observable for either of the
			// predicates here, it means an internal bug in
			// this package.  We avoid logging an error in
			// this case, because the SDK has to try its
			// own type conversion on the object.  The SDK
			// will see this and be forced to respond with
			// its own error.
			//
			// This code uses a double-nested if statement
			// to avoid creating a branch that is
			// impossible to cover.
			inst = floatObs
		}
	}
	return inst
}
```

(and same for int64). When I say the SDK is forced to check the same error in the comment above, I'm referring to [this](https://github.com/open-telemetry/opentelemetry-go/blob/81b2a33e1bbc57ef8f401e5ee7aaa9bc6b7ac772/sdk/metric/meter.go#L531).

Reply:
Makes sense, LGTM :+1: